exercises/IaC/ansible/conditionals/.templates/README.md.erb

```diff
@@ -0,0 +1,363 @@
+
+```
+
+<%= last_command_output%>
+```
+
+Alternatively, we can require both conditions to be met by using the `and` operator. Write the following YAML to `<%= and_condition_path = write_to_file('ansible/and_condition', example_and_yaml) %>`:
+
+```YAML
+
+<%= example_and_yaml %>
+```
+
+In this example we have met both conditions with values set against `day` and `time` so can expect the message to be printed.
+
+Run `<%= command("ansible-playbook #{and_condition_path} -c local")%>` to see that this happens.
+
+```
+
+<%= last_command_output%>
+```
+## Loops
+Often it will be necessary to perform the same action n times. For example creating new users in a database. This is where loops come in to play.
+
+### Simple loops
+
+Ansible provides the `loop` directive. It expects to be given a list of values. This list can be defined inline or can be the output of a [Jinja2 Expression](http://jinja.pocoo.org/docs/2.10/templates/#expressions).
+
+The following playbook defines a simple task that prints out a message for each item in the loop. The `loop` directive has been given the list of users identified in the `users` variable. By default Ansible will assign the current value which is the subject of a given iteration to a variable called `item`.
+
+```YAML
+
+<%=
+simple_loop =<<YAML
+---
+- name: 'create users'
+  hosts: all
+  vars:
+    users: [user1 user2 user3]
+  tasks:
+    - name: create users
+      debug: "creating user: {{item}}"
+      loop: "{{users}}"
+YAML
+%>
+```
+
+Write this example playbook to `<%= simple_loop_path = write_to_file('ansible/simple_loop.yml', simple_loop) %>` and run it with `<%= command("ansible-playbook #{simple_loop_path}") %>`.
+
+```
+
+<%= last_command_output %>
+```
+
+As you can see the message has been printinted for each of the values in the `users` list.
+
+### Looping through hashes: `dict2items`
+
+In some cases nested data structures such as hash will be the most appropriate structure for storing data. The following playbook shows a variable `user_groups` which holds maps users to the groups they belong to. In order to convert this hash in to a compatible list structure that can be passed to `loop` this is where the [Jinja Filter](https://docs.ansible.com/ansible/2.7/user_guide/playbooks_filters.html) [`dict2items`](https://docs.ansible.com/ansible/2.7/user_guide/playbooks_filters.html#dict-filter) comes in to play. The `dict2items` filter flattens a hash structure in to a list. The filter can be used as part of an expression like the following: `{{user_groups | dict2items}}`
+
+```YAML
+<%=
+hash_to_list=<<YAML
+- name: Hash to list example
+  hosts: all
+  vars:
+    user_groups:
+      Admin:
+      - user1
+      - user2
+      Team:
+      - user3
+      - user4
+      - user5
+
+  tasks:
+  - name: 'create user'
+    debug:
+      msg: "Creating user: {{item['key']}} in group: {{item['value']}}"
```

Comment:
This appears to be bass ackwards and gives an output of "Creating user: Admin in group: ['user1', 'user2']"